File: KthSmallest.java
public class KthSmallest {
    int cnt;
    int res;

    public int kthSmallest(TreeNode root, int k) {
        cnt = 0;
        res = 0;
        inorder(root, k);
        return res;
    }

    public void inorder(TreeNode node, int k) {
        if (node == null) {
            return;
        }
        inorder(node.left, k);

        if (cnt == k) {
            res = node.val;
            return;
        }
        inorder(node.right, k);
    }
}

File: KthSmallest.py
class Solution:
    def kthSmallest(self,root,k):
        self.kth=None
        self.k=k
        self.helper(root)

    def helper(self,node):
        if not node:
            return
        self.helper(node.left)
        self.k-=1
        if self.k == 0:
            self.res = node.val 
            return
        self.helper(node.right)
        
